api/v1/controllers/ProjectController.js

```diff
@@ -0,0 +1,165 @@
+	if(_.isNaN(page)){
+		var message = "Invalid page parameter";
+		var source = "page";
+		throw new errors.InvalidParameterError(message, source);
+	}
+	if(_.isNaN(count)){
+		var message = "Invalid count parameter";
+		var source = "count";
+		throw new errors.InvalidParameterError(message, source);
+	}
+	if(_.isNaN(published) || (published != 0 && published != 1)){
+		var message = "Invalid published parameter";
+		var source = "published";
+		throw new errors.InvalidParameterError(message, source);
+	}
+}
+
+function createProject (req, res, next) {
+	attributes = req.body;
+
+	PermissionService
+		.canCreateProject(req.user)
+		.then(function (isAuthed) {
+			return ProjectService.createProject(attributes);
+		})
+		.then(function (newProject) {
+			res.body = newProject.toJSON();
+
+			next();
+			return null;
+		})
+		.catch(function (error){
+			next(error);
+			return null;
+		});
+}
+
+function getProject (req, res, next) {
+	var id = req.params.id;
+
+	ProjectService
+		.findProjectById(id)
+		.then(function (project) {
+			res.body = project.toJSON();
+
+			next();
+			return null;
+		})
+		.catch(function (error) {
+			next(error);
+			return null;
+		});
+}
+
+function getAllProjects (req, res, next) {
+	var page = parseInt(req.params.page);
+	var count = parseInt(req.query.count);
+	var published = parseInt(req.query.published);
+
+	_validGetAllRequest(page, count, published);
+	
+	ProjectService
+		.getAllProjects(page, count, published)
+		.then(function (results) {
+			res.body = {};
+			res.body.projects = results;
+
+			next();
+			return null;
+		})
+		.catch(function (error) {
+			next(error);
+			return null;
+		});
+}
+
+function updateProject (req, res, next) {
+	var id = req.params.id;
+	var attributes = req.body;
+
+	ProjectService
+		.findProjectById(id)
+		.then(function (project) {
+			return ProjectService.updateProject(project, attributes);
+		})
+		.then(function (project) {
+			res.body = project.toJSON();
+
+			next();
+			return null;
+		})
+		.catch(function (error) {
+			next(error);
+			return null;
+		});
+}
+
+function addProjectMentor (req, res, next) {
+	var project_id = req.body.project_id;
+	var mentor_id = req.body.mentor_id;
+
+	ProjectService
+		.addProjectMentor(project_id, mentor_id)
+		.then(function (projectMentor) {
+			res.body = projectMentor.toJSON();
+
+			next();
+			return null;
+		})
+		.catch( function (error) {
+			next(error);
+			return null;
+		});
+}
+
+function deleteProjectMentor (req, res, next) {
+	var project_id = req.body.project_id;
+	var mentor_id = req.body.mentor_id;
+
+	ProjectService
+		.deleteProjectMentor(project_id, mentor_id)
+		.then(function () {
+			res.body = {}
+
+			next();
+			return null;
+		})
+		.catch( function (error) {
+			next(error);
+			return null;
+		});
+}
+
+router.use(bodyParser.json());
+router.use(middleware.auth);
+
+router.post('/mentor', middleware.permission(roles.ORGANIZERS), addProjectMentor);
```

Comment:
Hmmm, I'm only taking into 2 int values (representing ids), so I don't know if it's super necessary